src/proxy/http/add_header.rs

```diff
@@ -0,0 +1,229 @@
+use std::{fmt, marker::PhantomData};
+
+use http::header::{AsHeaderName, HeaderValue};
+
+use svc;
+
+type Ret<T> = fn(&T) -> Option<HeaderValue>;
+
+/// Wraps HTTP `Service` `Stack<T>`s so that a given header is removed from a
+/// request or response.
+#[derive(Clone)]
+pub struct Layer<H, T, R> {
+    header: H,
+    retrieve: Ret<T>,
+    _req_or_res: PhantomData<fn(R)>,
```

Comment:
Hm??? What does this mean?

This definitely falls within the realm of "surprisingly clever", in my opinion, and so it warrants an explanation.

Reply:
Note that this is based on code that already exists in the proxy (in the analogous `strip_header` layer): https://github.com/linkerd/linkerd2-proxy/blob/6cc755881994cba967619a0999dd7b2339c986e2/src/proxy/http/strip_header.rs#L12

I suggested that @kleimkuhler base his implementation off of that layer when we were discussing how to make this change...